src/sentry/runner/commands/presenters/consolepresenter.py

```diff
@@ -0,0 +1,102 @@
+    INVALID_TYPE_ERROR = "Option %s has invalid type. got %s, expected %s."
+
+    def __init__(self) -> None:
+        self.drifted_options: List[Tuple[str, str]] = []
+        self.channel_updated_options: List[str] = []
+        self.updated_options: List[Tuple[str, str, str]] = []
+        self.set_options: List[Tuple[str, str]] = []
+        self.unset_options: List[str] = []
+        self.error_options: List[Tuple[str, str]] = []
+        self.unregistered_options: List[str] = []
+        self.invalid_type_options: List[Tuple[str, str, str]] = []
+
+    def flush(self) -> None:
+        import logging
+
+        logger = logging.getLogger("sentry.options_automator")
+
+        for key, db_value in self.drifted_options:
+            click.echo(self.DRIFT_MSG % key)
+            logger.error(self.DRIFT_MSG % key)
+            if db_value != "":
+                click.echo(self.DB_VALUE % key)
+                # This is yaml instead of the python representation as the
+                # expected flow, in this case, is to use the output of this
+                # line to copy paste it in the config map.
+                click.echo(db_value)
+
+        for key in self.channel_updated_options:
+            click.echo(self.CHANNEL_UPDATE_MSG % key)
+
+        for key, db_value, value in self.updated_options:
+            click.echo(self.UPDATE_MSG % (key, db_value, value))
+
+        for key, value in self.set_options:
+            click.echo(self.SET_MSG % (key, value))
+
+        for key in self.unset_options:
+            click.echo(self.UNSET_MSG % key)
+
+        for key, reason in self.error_options:
+            click.echo(self.ERROR_MSG % (key, reason))
+
+        for key in self.unregistered_options:
+            click.echo(self.UNREGISTERED_OPTION_ERROR % key)
+            logger.error(self.UNREGISTERED_OPTION_ERROR, key)
+
+        for key, got_type, expected_type in self.invalid_type_options:
+            click.echo(self.INVALID_TYPE_ERROR % (key, got_type, expected_type))
+            logger.error(self.INVALID_TYPE_ERROR, key, got_type, expected_type)
+
+    def set(self, key: str, value: str) -> None:
+        self.set_options.append((key, value))
+
+    def unset(self, key: str) -> None:
+        self.unset_options.append(key)
+
+    def update(self, key: str, db_value: str, value: str) -> None:
+        self.updated_options.append((key, db_value, value))
+
+    def channel_update(self, key: str) -> None:
+        self.channel_updated_options.append(key)
+
+    def drift(self, key: str, db_value: str) -> None:
+        self.drifted_options.append((key, db_value))
+
+    def error(self, key: str, not_writable_reason: str) -> None:
+        self.error_options.append((key, not_writable_reason))
+
+    def unregistered(self, key: str) -> None:
+        self.unregistered_options.append(key)
+
+    def invalid_type(
+        self,
+        key: str,
+        got_type: str,
+        expected_type: str,
```

Comment:
nit: Please pass the types as `Type` object and not string. Passing them as string you are  basically formatting them for the presenter. The client code is not supposed to know how to format the Type object.